clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -82,10 +100,10 @@ public synchronized Cluster fetch() {
     }
 
     /**
-     * Add the topic to maintain in the metadata
+     * Add the topic to maintain in the metadata.
      */
     public synchronized void add(String topic) {
-        topics.add(topic);
+        topics.put(topic, null);
```

Comment:
I generally think that using null values in Maps is bad for two reasons:

* Many Map implementations don't support it (ConcurrentHashMap, for example)
* The commonly-used `get(foo) == null)` is no longer the same as `containsKey(foo)`, which can lead to difficult to spot bugs.

Why not use a sentinel like -1?